packages/rum-core/src/common/utils.js

```diff
@@ -34,33 +33,45 @@ function isCORSSupported() {
   return 'withCredentials' in xhr
 }
 
+/**
+ * Original source is from uuid module to strip `-` in the UUID generation logic
+ * https://github.com/uuidjs/uuid/blob/8977966d0061cca33a01a88f5b4893d3304d4840/src/bytesToUuid.js
+ */
 var byteToHex = []
 for (var i = 0; i < 256; ++i) {
   byteToHex[i] = (i + 0x100).toString(16).substr(1)
 }
 
-function bytesToHex(buf, offset) {
-  var i = offset || 0
-  var bth = byteToHex
-  // join used to fix memory issue caused by concatenation: https://bugs.chromium.org/p/v8/issues/detail?id=3175#c4
-  return [
-    bth[buf[i++]],
-    bth[buf[i++]],
-    bth[buf[i++]],
-    bth[buf[i++]],
-    bth[buf[i++]],
-    bth[buf[i++]],
-    bth[buf[i++]],
-    bth[buf[i++]],
-    bth[buf[i++]],
-    bth[buf[i++]],
-    bth[buf[i++]],
-    bth[buf[i++]],
-    bth[buf[i++]],
-    bth[buf[i++]],
-    bth[buf[i++]],
-    bth[buf[i++]]
-  ].join('')
+function bytesToHex(buffer) {
+  const hexOctets = []
+  for (let i = 0; i < buffer.length; i++) {
+    hexOctets.push(byteToHex[buffer[i]])
+  }
+  return hexOctets.join('')
+}
+
+/**
+ * Random number generator using crypto.getRandomValues
+ * uses msCrypto for supporting IE 11
+ * https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues
+ */
+const destination = new Uint8Array(16)
+
+function rng() {
+  if (typeof crypto != 'undefined' && crypto.getRandomValues) {
+    return crypto.getRandomValues(destination)
+  } else if (
+    typeof msCrypto != 'undefined' &&
+    typeof msCrypto.getRandomValues == 'function'
+  ) {
+    return msCrypto.getRandomValues(destination)
+  }
+  return destination
```

Comment:
I think here we shouldn't return `destination` if we can't generate one.

Reply:
its better to do this instead of undefined as our logic would still work for creating spans and transactions. We check the validity of the header here 
https://github.com/elastic/apm-agent-rum-js/blob/1501b11d6820cfbbaaa424320f75142777de3255/packages/rum-core/src/common/utils.js#L97-L103

Reply:
But we use the `generateRandomId` in a few different places (e.g. error id), even for spans and transactions the `isDtHeaderValid` only checks it if we should add the header or not. but the ids of could still be invalid.

Having said that, I'm not against having it as is (it might actually be useful for #249) but I think we'll need to check ids at some point to not send transaction/spans/errors with invalid ids.

Reply:
I guess, if the browser is not supported, we have to take another route and not even create/send any events. 

Yes, lets do this separately.